core/utils/eth_client_utils.go

```diff
@@ -2,29 +2,41 @@ package utils
 
 import (
 	"context"
-	"time"
-
 	"fmt"
+	"strings"
 
 	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
 	eigentypes "github.com/Layr-Labs/eigensdk-go/types"
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	connection "github.com/yetanotherco/aligned_layer/core"
 )
 
-const maxRetries = 25
-const sleepTime = 5 * time.Second
-
-func WaitForTransactionReceipt(client eth.InstrumentedClient, ctx context.Context, txHash gethcommon.Hash) (*types.Receipt, error) {
-	for i := 0; i < maxRetries; i++ {
-		receipt, err := client.TransactionReceipt(ctx, txHash)
+/*
+Errors:
+- "not found": (Transient) Call successfully returns but the tx receipt was not found.
+- "connect: connection refused": (Transient) Could not connect.
+*/
+func WaitForTransactionReceiptRetryable(client eth.InstrumentedClient, ctx context.Context, txHash gethcommon.Hash) (*types.Receipt, error) {
+	// For if no receipt and no error TransactionReceipt return "not found" as an error catch all ref: https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient.go#L313
+	receipt_func := func() (*types.Receipt, error) {
+		tx, err := client.TransactionReceipt(ctx, txHash)
 		if err != nil {
-			time.Sleep(sleepTime)
-		} else {
-			return receipt, nil
+			// Note return type will be nil
+			if err.Error() == "not found" {
+				return nil, connection.TransientError{Inner: err}
+			}
+			if strings.Contains(err.Error(), "connect: connection refused") {
+				return nil, connection.TransientError{Inner: err}
+			}
+			if strings.Contains(err.Error(), "read: connection reset by peer") {
+				return nil, connection.TransientError{Inner: err}
+			}
+			return nil, connection.PermanentError{Inner: fmt.Errorf("Permanent error: Unexpected Error while retrying: %s\n", err)}
```

Comment:
If we have a maxRetry, I think we should not throw permanent errors. Unless the transaction reverts.

Reply:
Actually, never mind, this will be handled in #1286.